src/test/kotlin/uk/gov/dluhc/printapi/service/PrintResponseFileReadinessServiceTest.kt

```diff
@@ -0,0 +1,123 @@
+        verify(sftpService).markFileForProcessing(DIRECTORY, matchingFileName2)
+        verify(printMessagingService).submitPrintResponseFileForProcessing(
+            ProcessPrintResponseFileMessage(DIRECTORY, renamedFileName1)
+        )
+        verify(printMessagingService).submitPrintResponseFileForProcessing(
+            ProcessPrintResponseFileMessage(DIRECTORY, renamedFileName2)
+        )
+        verifyNoMoreInteractions(sftpService, printMessagingService)
+    }
+
+    @Test
+    fun `should continue marking print response file and sending to queue when some of files marking fails due to exception`() {
+        // Given
+        val matchingFile1ThatThrowsException = "status-20221101171156051.json"
+        val matchingFile3ThatThrowsException = "status-20221101171156053.json"
+        val matchingFileName2 = "status-20221101171156052.json"
+        val matchingFileName4 = "status-20221101171156054.json"
+        val renamedFileName2 = "$matchingFileName2.processing"
+        val renamedFileName4 = "$matchingFileName4.processing"
+        val expectedExceptionThrown = RuntimeException("Some error")
+
+        given(sftpProperties.printResponseDownloadDirectory).willReturn(DIRECTORY)
+        given(sftpService.identifyFilesToBeProcessed(any())).willReturn(
+            listOf(
+                matchingFileName2,
+                matchingFile1ThatThrowsException,
+                matchingFileName4,
+                matchingFile3ThatThrowsException
+            )
+        )
+        given(sftpService.markFileForProcessing(eq(DIRECTORY), eq(matchingFile1ThatThrowsException))).willThrow(expectedExceptionThrown)
+        given(sftpService.markFileForProcessing(eq(DIRECTORY), eq(matchingFileName2))).willReturn(renamedFileName2)
+        given(sftpService.markFileForProcessing(eq(DIRECTORY), eq(matchingFile3ThatThrowsException))).willThrow(expectedExceptionThrown)
+        given(sftpService.markFileForProcessing(eq(DIRECTORY), eq(matchingFileName4))).willReturn(renamedFileName4)
+
+        // When
+        printResponseFileReadinessService.markPrintResponseFileForProcessing()
+
+        // Then
+        verify(sftpService).identifyFilesToBeProcessed(DIRECTORY)
+        verify(sftpService).markFileForProcessing(DIRECTORY, matchingFile1ThatThrowsException)
+        verify(sftpService).markFileForProcessing(DIRECTORY, matchingFileName2)
+        verify(sftpService).markFileForProcessing(DIRECTORY, matchingFile3ThatThrowsException)
+        verify(sftpService).markFileForProcessing(DIRECTORY, matchingFileName4)
+        verify(printMessagingService).submitPrintResponseFileForProcessing(
+            ProcessPrintResponseFileMessage(DIRECTORY, renamedFileName2)
+        )
+        verify(printMessagingService).submitPrintResponseFileForProcessing(
+            ProcessPrintResponseFileMessage(DIRECTORY, renamedFileName4)
+        )
+        verifyNoMoreInteractions(sftpService, printMessagingService)
```

Comment:
Notwithstanding my comments about the exception etc; this is an excellent test 👍 Nice one 👍

Reply:
Indeed the coverage is good and the test names are nice and clear, so I could understand what's being tested immediately 🙌